### DISCUSSION TITLE
Memory Leak with UploadFile and High Request Volume

### DISCUSSION THREAD

**Discussion by daxartio** [fastapi/fastapi#11290] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# fastapi
await request.form()

# my app
await request.read()
await request.close()

# memory leak
```

### Description

We have a service that receives a large number of files, each around 3-4 MB in size, at a rate of approximately 140 requests per second. We use UploadFile for file uploads.

We've encountered a problem where our service starts to leak memory. We suspect this is related to the lack of explicit request closure with await request.close(). We do call this method in our handler, but memory still leaks. It seems like an error might be occurring before the request is processed, even though await request.form() has already been executed.

We partially solved the problem by adding the following middleware:

```python
class FormCloserMiddleware(BaseHTTPMiddleware):
    async def dispatch(
        self,
        request: fastapi.Request,
        call_next: typing.Callable,
    ) -> fastapi.Response:
        try:
            return await call_next(request)
        finally:
            await request.close()
```

However, we suspect that this solution might not always work.

Is there a plan to add explicit closure on the FastAPI side? We believe this would be a great feature for developers, as most probably don't even know that files need to be closed. This lack of awareness led to the problem in our service.

### Operating System

Linux

### FastAPI Version

0.100.0

### Pydantic Version

1.10.7

### Python Version

3.10

**Comment by YuriiMotov** [comment]
Is it still relevant?
Let's close this for now to keep the discussions clean and organized.
If you still need help, feel free to let us know.
